WordPress/src/test/java/org/wordpress/android/util/wizard/WizardManagerTest.kt

```diff
@@ -0,0 +1,83 @@
+package org.wordpress.android.util.wizard
+
+import android.arch.core.executor.testing.InstantTaskExecutorRule
+import android.arch.lifecycle.Observer
+import com.nhaarman.mockitokotlin2.verify
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mock
+import org.mockito.junit.MockitoJUnitRunner
+import org.wordpress.android.ui.sitecreation.SiteCreationStep
+import org.wordpress.android.ui.sitecreation.SiteCreationStep.DOMAINS
+import org.wordpress.android.ui.sitecreation.SiteCreationStep.SEGMENTS
+import org.wordpress.android.ui.sitecreation.SiteCreationStep.SITE_PREVIEW
+
+private val STEPS = listOf(SEGMENTS, DOMAINS, SITE_PREVIEW)
+private val LAST_STEP_INDEX = STEPS.size - 1
+
+@RunWith(MockitoJUnitRunner::class)
+class WizardManagerTest {
+    @Rule
+    @JvmField val rule = InstantTaskExecutorRule()
+
+    private lateinit var manager: WizardManager<SiteCreationStep>
+    @Mock private lateinit var navigatorLiveDataObserver: Observer<SiteCreationStep>
+
+    @Before
+    fun setUp() {
+        manager = WizardManager(STEPS)
+        manager.navigatorLiveData.observeForever(navigatorLiveDataObserver)
+    }
+
+    @Test
+    fun `showNextStep is propagated`() {
+        manager.showNextStep()
+        verify(navigatorLiveDataObserver).onChanged(STEPS[0])
+    }
+
+    @Test
+    fun `showNextStep increments currentStepIndex`() {
+        val initialStepIndex = manager.currentStep
+        manager.showNextStep()
+        assertThat(manager.currentStep).isEqualTo(initialStepIndex + 1)
+    }
+
+    @Test
+    fun `step position is correctly calculated`() {
```

Comment:
Should we add a test for the `IllegalStateException` in `stepPosition` as well? Not really sure how much value it'd add, but I thought I'd bring it up 🤷‍♂️

Reply:
👍